### PR TITLE
timezone is corrected in event ics file, fixes #4860

### PR DESCRIPTION
**Pull request by jmcclelland**
Overview
----------------------------------------
As described in [the ticket](https://lab.civicrm.org/dev/core/-/issues/4860), the ICS file generated from the event page provides the time of the event using the UTC timezone instead of the local time zone, as specified in the Ical spec. This patch ensures the timezone is set properly.

Before
----------------------------------------
Previously, the generated ICS file's timezone section sets the DSTART item to UTC time, e.g. for an event that starts at 9:00 am America/New_York time, it would include:

```
BEGIN:VTIMEZONE
TZID:America/New_York
BEGIN:STANDARD
TZOFFSETFROM:-0500
TZOFFSETTO:-0500
TZNAME:EST
DTSTART:20240226T140000
END:STANDARD
END:VTIMEZONE

```

After
----------------------------------------

The timezone is properly set to local time, e.g.:

```
BEGIN:VTIMEZONE
TZID:America/New_York
BEGIN:STANDARD
TZOFFSETFROM:-0500
TZOFFSETTO:-0500
TZNAME:EST
DTSTART:20240226T090000
END:STANDARD
END:VTIMEZONE
```

Technical Details
----------------------------------------
The bug was introduced in the `date_create` function, which takes the actual timezone of the event as the second argument, not the aspirational timezone.

Comments
----------------------------------------
A test is included.

### CODE REVIEW

**Comment by wmortada** [comment]
Tsrr

**Comment by wmortada** [comment]
As far as I can see this test appears to be hard-coded to 4 Jan 2024 so I presume this will only work today (3 Jan 2024) but will fail every other day. Or have I misunderstood this?

File: tests/phpunit/CRM/Utils/ICalendarTest.php [hunk 1]
```diff
@@ -148,4 +148,42 @@ public function testSendWithAttachment(): void {
...
+    $this->eventCreateUnpaid($eventParameters);
+
+    $info = CRM_Event_BAO_Event::getCompleteInfo(NULL, NULL, $this->getEventId());
+    $calendar = explode("\n", CRM_Utils_ICalendar::createCalendarFile($info));
+    $expectedLines = [
+      "TZID:America/Los_Angeles" => FALSE,
+      "DTSTART:20240104T190000" => FALSE,
+      "DTSTAMP;TZID=America/Los_Angeles:20240104T190000" => FALSE,
+      "DTSTART;TZID=America/Los_Angeles:20240104T190000" => FALSE,
+      "DTEND;TZID=America/Los_Angeles:20240104T200000" => FALSE,
```
[the remark this answers is not in the document]
Comment (jmcclelland) [reply tests/phpunit/CRM/Utils/ICalendarTest.php]: Woops! Nothing like a second set of eyes to catch embarrasing mistakes. @wmortada thanks for the reality check. Should be fixed now.

**Comment by demeritcowboy** [comment]
I can confirm the problem in thunderbird and the patch fixes it.
I also tested outlook and google, which don't seem to care either way.

**Pull request merged**